### PR TITLE
Document the datacenter security model

### PR DESCRIPTION
**Pull request by cfrantz**
### Pull Request Overview

This document describes the Google datacenter security model. It describes how Google uses the [proprietary Titan chip](https://security.googleblog.com/2018/10/building-titan-better-security-through.html) in the datacenter and discusses some hypothetical use cases enabled by Tock's application identity model.

### Testing Strategy

N/A: Documentation only.

### TODO or Help Wanted

N/A

### Documentation Updated

- [ X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X ] Ran `make prepush`.

### CODE REVIEW

**Comment by alevy** [comment]
This is really cool, and super interesting to read! Thanks @cfrantz!

A small note that the text (while generally _highly_ readable and accessible) includes some left-over jargon: e.g. "prod" = "production server" or "production environment", "AP" = "application processor", etc.

These are probably second nature to many (including those of us who have spent enough time talking to folks at Google about this stuff), and probably easy enough to interpret for many more, but seems like clarity is worth a few extra characters in this case.

File: doc/wg/opentitan/case_studies/datacenter_security_model.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+Titan integrations into server peripherals are very similar to the integration into a server product: Titan is given low-level control over the peripheral's reset signals and is positioned such that it has control over the peripheral's boot firmware.  There are often customizations to the integration to meet certain requirements of the peripheral (such as boot timing), but for the purpose of this document, the integrations are basically the same.
+
+
+## Titan Code Images
+
+Titan has 3 distinct code images. The first, called the ROM image, never changes. The second, called the bootloader, changes rarely. The third, called the application firmware,
+
+is a monolithic image including the kernel, hardware support code and application code.  Generally, new application firmware is pushed to prod every few months.
```
Comment (brghena) [inline doc/wg/opentitan/case_studies/datacenter_security_model.md:48]: Weird line break here.

File: doc/wg/opentitan/case_studies/datacenter_security_model.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+The Google Titan chip is a Google-designed Root-of-Trust chip.  It features a 32-bit embedded ARM Cortex-M, internal flash and RAM and crypto acceleration hardware, including a SHA hashing block, a key derivation block and a bignum accelerator.
+
+
+## How Titan is Integrated into the Datacenter
+
+The Google Titan chip is integrated into server products such that it has control over the machine's reset process and boot firmware.  Specifically, the Titan chip can both monitor and drive the application processor's reset signals and interposes on the SPI bus between the AP and its EEPROM. This design is applied to simple servers with a CPU, to BMCs, and to peripherals like NICs and accelerators.
```
Comment (brghena) [inline doc/wg/opentitan/case_studies/datacenter_security_model.md:39]: Not sure what an "AP" or "BMC" are in this context. It's probably worth spelling them out at least once.

**Review by brghena — COMMENTED** [review]
This is really interesting information and well-written. Thanks for this!

File: doc/wg/opentitan/case_studies/datacenter_security_model.md (hunk 1 above)
Comment (cfrantz) [reply doc/wg/opentitan/case_studies/datacenter_security_model.md]: Fixed.

File: doc/wg/opentitan/case_studies/datacenter_security_model.md (hunk 2 above)
Comment (cfrantz) [reply doc/wg/opentitan/case_studies/datacenter_security_model.md]: Added definitions.

**Comment by cfrantz** [comment]
I've added a second commit that reformats the document from the raw output of the Google-Docs-to-Markdown converter tool to a document that can also be read as plain text.

**Comment by hudson-ayers** [comment]
The netlify failures here are unrelated to this PR; netlify just deprecated ubuntu 16.04 support. Working on a fix.

**Comment by hudson-ayers** [comment]
> The netlify failures here are unrelated to this PR; netlify just deprecated ubuntu 16.04 support. Working on a fix.

I think that it should be resolved now, but you'll have to force-push the PR for the netlify build to retry (unlike with the github actions CI I don't know how to force netlify to retry this specific PR and push updates to the build status)

**Comment by lschuermann** [comment]
> you'll have to force-push the PR for the netlify build to retry (unlike with the github actions CI I don't know how to force netlify to retry this specific PR and push updates to the build status)

I think the following should do the trick (at least to indicate whether the current state of the PR would pass CI):

bors try

**Review by hudson-ayers — DISMISSED** [review]

**Review by lschuermann — DISMISSED** [review]
Is there anything missing from transitioning this out of draft status?

**Review by hudson-ayers — APPROVED** [review]

**Review by lschuermann — APPROVED** [review]

**Review by alevy — APPROVED** [review]

**Pull request merged**